### PR TITLE
Adding DFS and ToT-crosswords example

### PR DESCRIPTION
**Pull request by mhd0528**

### CODE REVIEW

File: examples/tot_crosswords/__init__.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+from world_model import game24State, game24Action, game24WorldModel
```
Comment (Ber666) [inline examples/tot_crosswords/__init__.py:1]: Better to capitalize these class names?

File: examples/tot_crosswords/example.py
Comment (Ber666) [inline examples/tot_crosswords/example.py:1]: What's this file?

File: examples/tot_crosswords/inference.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+from reasoners import LanguageModel, Reasoner, SearchAlgorithm
+from reasoners.algorithm import DFS
+
+from world_model import crosswordsWorldModel
+from search_config import crosswordsConfig
+from utils import MiniCrosswordsEnv
+
+
+
+def rap_crosswords(base_model: LanguageModel,
```
Comment (Ber666) [inline examples/tot_crosswords/inference.py:17]: Let's not say it's RAP. It's reproducing ToT

File: examples/tot_crosswords/search_config.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import numpy as np
+import re
+
+from reasoners import SearchConfig, LanguageModel
+from world_model import crosswordsState, crosswordsAction
+from utils import *
+from prompts.crosswords import * 
+
+
+class crosswordsConfig(SearchConfig):
```
Comment (Ber666) [inline examples/tot_crosswords/search_config.py:12]: Capitalize the class name?

File: examples/tot_crosswords/prompts/crosswords.py [hunk 4]
```diff
@@ -0,0 +1,326 @@
+# 5 shot
```
Comment (Ber666) [inline examples/tot_crosswords/prompts/crosswords.py:1]: What's the difference from previous json file?

File: reasoners/lm/openai_model.py [hunk 5]
```diff
@@ -48,7 +53,17 @@ def generate(self,
                 ### GPT 3.5 and higher use a different API
                 if ('gpt-3.5' in self.model) or ('gpt-4' in self.model):
                     messages = [{"role": "user", "content": prompt}]
-                    response = openai.ChatCompletion.create(
+                    # response = openai.ChatCompletion.create(
```
Comment (Ber666) [inline reasoners/lm/openai_model.py:56]: remove useless comment

**Comment by Ber666** [comment]
Did you update the readme to guide users to reproduce the results?

File: examples/tot_crosswords/utils.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
+import re
+import json
+import os
+from prompts.crosswords import * 
+
+class MiniCrosswordsEnv:
+    def __init__(self, file='mini0505.json'):
```
Comment (Ber666) [inline examples/tot_crosswords/utils.py:7]: what's this file? did you mention it in readme?

File: examples/tot_crosswords/prompts/crosswords.py (hunk 4 above)
Comment (mhd0528) [reply examples/tot_crosswords/prompts/crosswords.py:1]: we used this one for prompts. I removed the json file

**Comment by mhd0528** [comment]
I have removed unused files and comments, updated readme, and changed those class names.

**Pull request merged**